src/telliot/contract/contract.py

```diff
@@ -184,19 +183,33 @@ async def write_with_retry(
                     acc_nonce += 1
                 elif not status.ok and status.error and "nonce too low" in status.error:
                     acc_nonce += 1
+                # a different rpc error
+                elif (
```

Comment:
Not necessarily a now thing, but in the future it would be nice to have some more comments about the specifics of the what the error is about. Maybe point to some explainer in the docs about typical failures.

Reply:
I agree